ts/components/messages/paginated/hooks/useMessageOpening.tsx

```diff
@@ -0,0 +1,55 @@
+import { useNavigation } from "@react-navigation/native";
+import { useCallback } from "react";
+import { UIMessage } from "../../../../store/reducers/entities/messages/types";
+import ROUTES from "../../../../navigation/routes";
+import { TagEnum } from "../../../../../definitions/backend/MessageCategoryPN";
+import { usePnOpenConfirmationBottomSheet } from "../../../../features/pn/components/PnOpenConfirmationBottomSheet";
+import { pnEnabled } from "../../../../config";
+
+export const useMessageOpening = () => {
+  const navigation = useNavigation();
+
+  const navigate = useCallback(
+    (message: UIMessage) => {
+      navigation.navigate(ROUTES.MESSAGES_NAVIGATOR, {
+        screen: ROUTES.MESSAGE_ROUTER_PAGINATED,
+        params: {
+          messageId: message.id
+        }
+      });
+    },
+    [navigation]
+  );
+
+  const pnBottomSheet = usePnOpenConfirmationBottomSheet({
+    onConfirm: (message: UIMessage, _: boolean) => {
+      pnBottomSheet.dismiss();
```

Comment:
I think that both these dismiss functions may be called inside the `usePnOpenConfirmationBottomSheet` avoiding this redundant call, since the dismiss function should be available inside the main hook for the bottomsheet. What do you think about this solution?

Reply:
Fixed 145baa301625f7d2cd1c1eb22b359df4b23fad9e. I've also removed the `onCancel` callback as it's not needed anymore.